### PR TITLE
stop using bare ROOT objects in `HLTDQMHist*` classes

### PR DESCRIPTION
**Pull request by missirol**
#### PR description:

This PR tries to implement the suggestion in https://github.com/cms-sw/cmssw/issues/40676#issuecomment-1428768169, to avoid accessing bare ROOT objects (via `MonitorElement`) in the `HLTDQMHist*D` classes of the HLT offline DQM.

Minor cleanup is also applied to `HLTTagAndProbeOfflineSource<..>::fillDescriptions`.

Merely technical. No changes expected.

FYI: @Sam-Harper

#### PR validation:

`runTheMatrix.py -l 11634.0` passed.

#### If this PR is a backport, please specify the original PR and why you need to backport that PR. If this PR will be backported, please specify to which release cycle the backport is meant for:

To be backported to ~`CMSSW_13_0_X` if merged~ down to `CMSSW_12_4_X`.

### CODE REVIEW

File: DQMOffline/Trigger/plugins/HLTTagAndProbeOfflineSource.cc [hunk 1]
```diff
@@ -51,16 +38,10 @@ template <typename TagType, typename TagCollType, typename ProbeType, typename P
...
                 HLTDQMTagAndProbeEff<TagType, TagCollType, ProbeType, ProbeCollType>::makePSetDescription(),
-                std::vector<edm::ParameterSet>());
-
-  // addDefault must be used here instead of add unless this function is specialized
-  // for different sets of template parameter types. Each specialization would need
-  // a different module label. Otherwise the generated cfi filenames will conflict
-  // for the different plugins.
-  descriptions.addDefault(desc);
+                {});
+  descriptions.addWithDefaultLabel(desc);
```
Comment (missirol) [inline DQMOffline/Trigger/plugins/HLTTagAndProbeOfflineSource.cc:44]: This change is unrelated to the main goal of this PR. It correctly creates the following `cfi` files (one per template instantiation):
```bash
"${CMSSW_BASE}"/cfipython/"${SCRAM_ARCH}"/DQMOffline/Trigger/hltElePhoTagAndProbeOfflineSource_cfi.py
"${CMSSW_BASE}"/cfipython/"${SCRAM_ARCH}"/DQMOffline/Trigger/hltEleTagAndProbeOfflineSource_cfi.py
"${CMSSW_BASE}"/cfipython/"${SCRAM_ARCH}"/DQMOffline/Trigger/hltMuEleTagAndProbeOfflineSource_cfi.py
"${CMSSW_BASE}"/cfipython/"${SCRAM_ARCH}"/DQMOffline/Trigger/hltMuPhoTagAndProbeOfflineSource_cfi.py
"${CMSSW_BASE}"/cfipython/"${SCRAM_ARCH}"/DQMOffline/Trigger/hltMuTagAndProbeOfflineSource_cfi.py
"${CMSSW_BASE}"/cfipython/"${SCRAM_ARCH}"/DQMOffline/Trigger/hltPhoTagAndProbeOfflineSource_cfi.py
```

File: DQMOffline/Trigger/plugins/HLTTagAndProbeOfflineSource.cc [hunk 2]
```diff
@@ -51,16 +38,10 @@ template <typename TagType, typename TagCollType, typename ProbeType, typename P
 void HLTTagAndProbeOfflineSource<TagType, TagCollType, ProbeType, ProbeCollType>::fillDescriptions(
     edm::ConfigurationDescriptions& descriptions) {
   edm::ParameterSetDescription desc;
-  desc.add<edm::InputTag>("objs", edm::InputTag(""));
```
Comment (missirol) [inline DQMOffline/Trigger/plugins/HLTTagAndProbeOfflineSource.cc:54]: This does not seem to be used, so it is removed.

**Comment by missirol** [comment]
please test

**Comment by Sam-Harper** [comment]
thanks Marino, I wasnt aware of the original comment. Much appreciated for you taking care of it.

**Comment by missirol** [comment]
@cms-sw/dqm-l2 , could you please review this PR?

**Comment by emanueleusai** [comment]
+1

**Comment by perrotta** [comment]
+1
